### ISSUE TITLE
Visual Shader - Clamp Zoom to 30-100%

### ISSUE THREAD

**Issue by Norrox** [godotengine/godot-proposals#7368]
### Describe the project you are working on

Alot of random projects

### Describe the problem or limitation you are having in your project

Not a problem really, mostly an annoyance. When you are zoomed out and need to zoom in you mostly zoom past the 100% mark, and have to press the 1:1 button

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Lock the zoom to 30-100%, is there really a need for it to go past 100%?
![Skärmbild 2023-07-25 103936](https://github.com/godotengine/godot-proposals/assets/13393087/d841c7fb-13a2-4547-88cc-eb440d6eb370)
![Skärmbild 2023-07-25 103930](https://github.com/godotengine/godot-proposals/assets/13393087/0236233d-e830-4240-a6cc-abf50b9c75aa)

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

When you scroll in to zoom in it stops att 100%

### If this enhancement will not be used often, can it be worked around with a few lines of script?

All the time i think?

### Is there a reason why this should be core and not an add-on in the asset library?

Should be in core.

**Comment by Calinou** [comment]
I wouldn't clamp zoom to 100%, as this is an accessibility concern for users with low vision or when doing screencasts/presentations.

**Comment by Norrox** [comment]
Can't it be an checkbox in the editor settings? Accessibility Mode [ x ]
? :)

**Comment by Zireael07** [comment]
I agree that having to zoom past 100% is rarely needed (even for someone using glasses as I do), and that getting back to 100% can be difficult.
I'd make it a checkbox next to zoom buttons not editor settings though.

**Comment by Calinou** [comment]
> I'd make it a checkbox next to zoom buttons not editor settings though.

I'm not aware of any application that has a "clamp zoom" checkbox right next to its zoom controls. This would take up a lot of space for something you don't toggle often.

**Comment by Zireael07** [comment]
I'm not aware of any either, but hiding this in editor settings is NOT the way to go imho.

+100 to having an easily accessible keyboard shortcut to reset zoom

**Comment by sammonius** [comment]
An easier solution would be just to add a shortcut for the 1:1 button. Maybe one of the F# keys.

**Comment by Calinou** [comment]
> EDIT: Or make zooming towards 100% go faster than zooming away from it.

This means that the mouse wheel would behave differently depending on direction, which I rarely find to be a good idea.
